Source/DataSources/I3SDataSource.js

```diff
@@ -0,0 +1,4601 @@
+        geometryData.vertexCount,
+        geometryData.uv0s,
+        geometryData["uv-region"]
+      );
+    }
+
+    // Create the final buffer
+    var meshData = generateGLTFBuffer(
+      geometryData.vertexCount,
+      geometryData.indices,
+      geometryData.positions,
+      geometryData.normals,
+      geometryData.uv0s,
+      geometryData.colors
+    );
+
+    var customAttributes = {};
+    if (geometryData["feature-index"]) {
+      customAttributes.positions = geometryData.positions;
+      customAttributes.indices = geometryData.indices;
+      customAttributes["feature-index"] = geometryData["feature-index"];
+      customAttributes.cartesianCenter = e.data.cartesianCenter;
+      customAttributes.parentRotation = e.data.parentRotation;
+    } else if (geometryData["faceRange"]) {
+      customAttributes.positions = geometryData.positions;
+      customAttributes.indices = geometryData.indices;
+      customAttributes.sourceURL = e.data.url;
+      customAttributes.cartesianCenter = e.data.cartesianCenter;
+      customAttributes.parentRotation = e.data.parentRotation;
+
+      //Build the feature index array from the faceRange. This should store the
+      customAttributes["feature-index"] = new Array(
+        geometryData.positions.length
+      );
+      for (
+        var range = 0;
+        range < geometryData["faceRange"].length - 1;
+        range += 2
+      ) {
+        var curIndex = range / 2;
+        var rangeStart = geometryData["faceRange"][range];
+        var rangeEnd = geometryData["faceRange"][range + 1];
+        for (var i = rangeStart; i <= rangeEnd; i++) {
+          customAttributes["feature-index"][i * 3] = curIndex;
+          customAttributes["feature-index"][i * 3 + 1] = curIndex;
+          customAttributes["feature-index"][i * 3 + 2] = curIndex;
+        }
+      }
+    }
+
+    meshData.customAttributes = customAttributes;
+    postMessage(meshData);
+  };
+}
+
+/*
+.########..########...#######...######..########..######...######..####.##....##..######..
+.##.....##.##.....##.##.....##.##....##.##.......##....##.##....##..##..###...##.##....##.
+.##.....##.##.....##.##.....##.##.......##.......##.......##........##..####..##.##.......
+.########..########..##.....##.##.......######....######...######...##..##.##.##.##...####
+.##........##...##...##.....##.##.......##.............##.......##..##..##..####.##....##.
+.##........##....##..##.....##.##....##.##.......##....##.##....##..##..##...###.##....##.
+.##........##.....##..#######...######..########..######...######..####.##....##..######..
+
+..#######..##.....##.########.##.....##.########
+.##.....##.##.....##.##.......##.....##.##......
+.##.....##.##.....##.##.......##.....##.##......
+.##.....##.##.....##.######...##.....##.######..
+.##..##.##.##.....##.##.......##.....##.##......
+.##....##..##.....##.##.......##.....##.##......
+..#####.##..#######..########..#######..########
+*/
+function I3SGLTFProcessingQueue() {
```

Comment:
Ideally, all worker related code should use [`TaskProcessor`](https://github.com/CesiumGS/cesium/blob/main/Source/Core/TaskProcessor.js).